keras_cv/models/object_detection/yolo_v8/yolo_v8_backbone.py

```diff
@@ -0,0 +1,167 @@
+)
+from keras_cv.models.object_detection.yolo_v8.yolo_v8_layers import (
+    apply_csp_block,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+def apply_spatial_pyramid_pooling_fast(
+    inputs, pool_size=5, activation="swish", name=None
+):
+    channel_axis = -1
+    input_channels = inputs.shape[channel_axis]
+    hidden_channels = int(input_channels // 2)
+
+    nn = apply_conv_bn(
+        inputs,
+        hidden_channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_pre",
+    )
+    pool_1 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool1"
+    )(nn)
+    pool_2 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool2"
+    )(pool_1)
+    pool_3 = layers.MaxPool2D(
+        pool_size=pool_size, strides=1, padding="same", name=f"{name}_pool3"
+    )(pool_2)
+
+    out = tf.concat([nn, pool_1, pool_2, pool_3], axis=channel_axis)
+    out = apply_conv_bn(
+        out,
+        input_channels,
+        kernel_size=1,
+        activation=activation,
+        name=f"{name}_output",
+    )
+    return out
+
+
+# TODO(ianstenbit) This should probably just be a CSPDarknet
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class YOLOV8Backbone(Backbone):
+    def __init__(
+        self,
+        include_rescaling,
+        channels=[128, 256, 512, 1024],
+        depths=[3, 6, 6, 3],
+        input_shape=(512, 512, 3),
+        activation="swish",
+        **kwargs,
+    ):
+        inputs = layers.Input(input_shape)
+
+        x = inputs
+        if include_rescaling:
+            x = layers.Rescaling(1 / 255.0)(x)
+
+        """ Stem """
+        # stem_width = stem_width if stem_width > 0 else channels[0]
+        stem_width = channels[0]
+        x = apply_conv_bn(
+            x,
+            stem_width // 2,
+            kernel_size=3,
+            strides=2,
+            activation=activation,
+            name="stem_1",
+        )
+        x = apply_conv_bn(
+            x,
+            stem_width,
+            kernel_size=3,
+            strides=2,
+            activation=activation,
+            name="stem_2",
+        )
+
+        """ blocks """
+        pyramid_level_inputs = {1: x.node.layer.name}
+        for stack_id, (channel, depth) in enumerate(zip(channels, depths)):
+            stack_name = f"stack{stack_id + 1}"
+            if stack_id >= 1:
+                x = apply_conv_bn(
+                    x,
+                    channel,
+                    kernel_size=3,
+                    strides=2,
+                    activation=activation,
+                    name=f"{stack_name}_downsample",
+                )
+            x = apply_csp_block(
+                x,
+                depth=depth,
+                expansion=0.5,
+                activation=activation,
+                name=f"{stack_name}_c2f",
+            )
+
+            if stack_id == len(depths) - 1:
```

Comment:
Why not just remove this from the loop if it happens at the end anyway?

Reply:
Well it's a little hairy either way. If we take it out of the loop, the logic for setting `pyramid_level_inputs` gets a bit awkward because you have to special-case it for the last stack, where we'd set the output outside of the loop.

IMO it's a bit cleaner this way but I'm happy to break it up.